src/project-converter.ts

```diff
@@ -0,0 +1,159 @@
+    try {
+      this.conversionSettings.includes.has(document.url) ?
+          this.convertDocumentToJs(document, new Set()) :
+          this.convertDocumentToHtml(document, new Set());
+    } catch (e) {
+      console.error(`Error in ${document.url}`, e);
+    }
+  }
+
+  /**
+   * Check if a document is explicitly excluded or has already been converted
+   * to decide if it should be converted or skipped.
+   */
+  shouldConvertDocument(document: Document): boolean {
+    const documentUrl = getDocumentUrl(document);
+    if (this.conversionResults.has(documentUrl)) {
+      return false;
+    }
+    for (const exclude of this.conversionSettings.excludes) {
+      if (documentUrl.endsWith(exclude)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Specifically convert an HTML document to a JS module. Useful during
+   * conversion for dependencies where the type of result is explictly expected.
+   */
+  convertDocumentToJs(document: Document, visited: Set<OriginalDocumentUrl>) {
+    if (!this.shouldConvertDocument(document)) {
+      return;
+    }
+    const documentConverter = new DocumentConverter(this, document, visited);
+    const newModule = documentConverter.convertToJsModule();
+    this.handleConversionResult(newModule);
+  }
+
+  /**
+   * Specifically convert an HTML document without changing the file type
+   * (changes imports and inline scripts to modules as necessary.) Useful during
+   * conversion for dependencies where the type of result is explictly expected.
+   */
+  convertDocumentToHtml(document: Document, visited: Set<OriginalDocumentUrl>) {
+    if (!this.shouldConvertDocument(document)) {
+      return;
+    }
+    const documentConverter = new DocumentConverter(this, document, visited);
+    const newModule = documentConverter.convertAsToplevelHtmlDocument();
+    this.handleConversionResult(newModule);
+  }
+
+  /**
+   * A private instance method for handling new conversion results, exports,
+   * etc.
+   */
+  private handleConversionResult(newModule: ConversionResult): void {
+    this.conversionResults.set(newModule.originalUrl, newModule);
+    if (newModule.output.type === 'js-module') {
+      for (const expr of newModule.output.exportedNamespaceMembers) {
+        this.namespacedExports.set(
+            expr.oldNamespacedName,
+            new JsExport(newModule.convertedUrl, expr.es6ExportName));
+      }
+    }
+  }
+
+  /**
+   * This method collects the results after all documents are converted. It
+   * handles out some broken edge-cases (ex: shadycss) and sets empty map
+   * entries for files to be deleted.
+   */
+  getResults(): Map<ConvertedDocumentFilePath, string|undefined> {
+    const results = new Map<ConvertedDocumentFilePath, string|undefined>();
+
+    for (const convertedModule of this.conversionResults.values()) {
+      if (convertedModule.originalUrl.endsWith(
+              'shadycss/entrypoints/apply-shim.js') ||
```

Comment:
Maybe this can be `/shadycss/entrypoints/apply-shim.js` in case I have an element using `notshadycss`?  Far fetched?

Reply:
in workspace mode their originalUrl is literally `"shadycss/entrypoints/apply-shim.js"`. But if we never write this, maybe we should just add it to excludes? Not sure why this is being handled seperately.

Reply:
Oh right, we want to preserve any imports of this file (excludes completely removes/ignores them). 

I'm going to comment this and mark as TODO for now, and we can revisit after this PR. Current logic is *okay*, and it's unclear if this list will need to be user-definable at some point or can live as a hardcoded one-off.